### PR TITLE
DA-109 Create backend engine for automated BC creation

### PR DESCRIPTION
**Pull request by janinko**
There is still one TODO in the code: https://github.com/project-ncl/dependency-analysis/pull/143/files#diff-7ace98b084b94613d03fb4155ed18d62R89
We must create Product in PNC somehow and obtain productId.

### CODE REVIEW

File: bc-backend/src/main/java/org/jboss/da/bc/backend/api/RepoClonator2000.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package org.jboss.da.bc.backend.api;
+
+/**
+ *
+ * @author Honza Brázdil <jbrazdil@redhat.com>
+ */
+public interface RepoClonator2000 {
```
Comment (jbartece) [inline bc-backend/src/main/java/org/jboss/da/bc/backend/api/RepoClonator2000.java]: Good joke, but rebase it after https://github.com/project-ncl/dependency-analysis/pull/145 will be merged

**Comment by jbartece** [comment]
retest this please

**Comment by jbartece** [comment]
Looks OK

File: reports-backend/src/main/java/org/jboss/da/reports/backend/impl/DependencyTreeGeneratorImpl.java [hunk 2]
```diff
@@ -12,11 +13,13 @@
...
 import javax.enterprise.context.ApplicationScoped;
 import javax.inject.Inject;
+import org.apache.maven.scm.ScmException;
+import org.jboss.da.communication.scm.api.SCMConnector;
 
 /**
  *
  * @author Dustin Kut Moy Cheung <dcheung@redhat.com>
- *
+ * @author Honza Brázdil <janinko.g@gmail.com>
```
Comment (jomrazek) [inline reports-backend/src/main/java/org/jboss/da/reports/backend/impl/DependencyTreeGeneratorImpl.java]: maybe you want here your redhat account

**Comment by jomrazek** [comment]
:+1:

**Comment by jbartece** [comment]
retest this please

**Comment by janinko** [comment]
Just the last 3 commits. I'll rebase when they are merged.

**Comment by jbartece** [comment]
:+1:

**Pull request merged**